pennylane/devices/null_qubit.py

```diff
@@ -11,320 +11,374 @@
-            "PauliY": self._apply_y,
-            "PauliZ": self._apply_z,
-            "Hadamard": self._apply_hadamard,
-            "S": self._apply_s,
-            "T": self._apply_t,
-            "SX": self._apply_sx,
-            "CNOT": self._apply_cnot,
-            "SWAP": self._apply_swap,
-            "CZ": self._apply_cz,
-            "Toffoli": self._apply_toffoli,
-        }
-
-    # pylint: disable=arguments-differ
-    def apply(self, operations, *args, **kwargs):
-        for op in operations:
-            self._apply_operation(self._state, op)
-
-    def _apply_operation(self, state, operation):
-        self._operation_calls[operation.name] += 1
-
-        if operation.name in self._apply_ops:
-            return self._apply_ops[operation.name](state, axes=None)
-
-        wires = operation.wires
-        if operation in diagonal_in_z_basis:
-            return self._apply_diagonal_unitary(state, None, wires)
-        if len(wires) <= 2:
-            # Einsum is faster for small gates
-            return self._apply_unitary_einsum(state, None, wires)
-        return self._apply_unitary(state, None, wires)
-
-    def _apply_x(self, state, axes, **kwargs):
-        return [0.0]
-
-    def _apply_y(self, state, axes, **kwargs):
-        return [0.0]
-
-    def _apply_z(self, state, axes, **kwargs):
-        return [0.0]
-
-    def _apply_hadamard(self, state, axes, **kwargs):
-        return [0.0]
-
-    def _apply_s(self, state, axes, inverse=False):
-        return [0.0]
-
-    def _apply_t(self, state, axes, inverse=False):
-        return [0.0]
-
-    def _apply_sx(self, state, axes, inverse=False):
-        return [0.0]
-
-    def _apply_cnot(self, state, axes, **kwargs):
-        return [0.0]
-
-    def _apply_swap(self, state, axes, **kwargs):
-        return [0.0]
-
-    def _apply_cz(self, state, axes, **kwargs):
-        return [0.0]
-
-    def _apply_toffoli(self, state, axes, **kwargs):
-        return [0.0]
-
-    def _apply_phase(self, state, axes, parameters, inverse=False):
-        return [0.0]
-
-    def expval(self, observable, shot_range=None, bin_size=None):
-        return [0.0]
-
-    def var(self, observable, shot_range=None, bin_size=None):
-        return [0.0]
-
-    @classmethod
-    def capabilities(cls):
-        capabilities = super().capabilities().copy()
-        capabilities.update(
-            model="qubit",
-            supports_inverse_operations=True,
-            supports_analytic_computation=True,
-            supports_broadcasting=False,
-            returns_state=True,
-            passthru_devices={
-                "tf": "null.qubit",
-                "torch": "null.qubit",
-                "autograd": "null.qubit",
-                "jax": "null.qubit",
-            },
-        )
-        return capabilities
+    .. code-block:: python
 
-    @staticmethod
-    def _create_basis_state(index):
-        return [0.0]
+        n_layers = 5
+        n_wires = 10
```

Comment:
Might be worth bumping this up to `100` to showcase that you can use it for large circuits that can't currently be used with actual devices.